src/utilities/get-issue-data.ts

```diff
@@ -0,0 +1,19 @@
+import { Octokit } from '@octokit/rest';
+import { IssueDef } from '../APIs/issue-def';
+
+export function getIssueData(issueNumber: number): Promise<IssueDef> {
```

Comment:
So I figured out that using the [@octokit/type](https://www.npmjs.com/package/@octokit/types) provides all the types we need, however using it is **interesting**. I can see why its in another library.

This is this full fill I created that removed the current manual `IssueDef` and provides first part type support for current and future endpoints.

I also refactored to code to rely more on inferred types with full TS coverage:
```typescript
import { Octokit } from '@octokit/rest';
import { Endpoints } from '@octokit/types';

// This can be refactored to what we needed, or made more specific,
// or renamed to be more clear? This also can be moved to a common file.
export type IssueEndpoint = Endpoints['GET /repos/:owner/:repo/issues/:issue_number'];

export const getIssueData = (
  issueNumber: number
): Promise<IssueEndpoint['response']> =>
  new Octokit().request('GET /repos/{owner}/{repo}/issues/{issue_number}', {
    owner: 'freeCodeCamp',
    repo: 'freeCodeCamp',
    issue_number: issueNumber
  });
```

Reply:
Added types from the package. Thanks Brad!